flow/flow.go

```diff
@@ -94,10 +94,18 @@ type RawPackets struct {
 	RawPackets []*RawPacket
 }
 
+type LayerKeyMode int
+
+const (
+	L2L3KeyMode LayerKeyMode = 0 // uses Layer2 and Layer3 for hash computation, default mode
```

Comment:
Hi Sylvain,

Thanks for adding this so quickly :-)

The obvious question, why L2L3KeyMode is default:: 

L2L3KeyMode is not the expected behavior for regular users working on networks. I agree that having the option to always take into account L2 is useful for network experts that understand L2 layer and the fact that L3 traffic can move on different L2. But when you say network flow to people ... most will assume L3 (IP) and most will not want to know about the L2 options that traffic can pass through. Also it makes things more complicated when using  persistence storage, you need to aggregate all L2 flows to get L3 layer info.

Your change to the key that takes into account L2 when L3 doesn't exist makes L3KeyMode very compelling almost to the point that I think that there is no real practical need for the L2L3KeyMode.

One more small idea ... when in L3KeyMode it can be nice to add a small icon, or label in the flow table (UI) that show which key type is used, so user will have clarity on what is a flow L2 based or L3 based.

Reply:
I think that the l2L3 is useful as Skydive is a debugging tool too and in a multipath context you may want to have one flow per path to ensure that the traffic is correctly forwarded. 

While writing this I thought making L3 by default or adding a configuration option to change the default ?